### ISSUE TITLE
RabbitMQ source hangs when message cannot be processed and retry configuration is set

### ISSUE THREAD

**Issue by gunpuz** [knative-sandbox/eventing-rabbitmq#830]
**Describe the bug**

Architecture:
`RabbitMQ` -> `RabbitMQ event source` -> `HTTP service that returns error`

Seems that when retry is configured, then exception occours
<img width="617" alt="image" src="https://user-images.githubusercontent.com/39419120/175068623-5e070ad3-ff5f-4d68-a7b7-4ec462be9032.png">

If retry values are not configured, then error is written in rabbitmq source log and next messages are processed

##### Example rabbitmq source 
```yaml
apiVersion: sources.knative.dev/v1alpha1
kind: RabbitmqSource
metadata:
  name: my-event-source
spec:
  broker: "rabbitmq:5672/"
  connectionSecret:
    name: rabbitmq
  user:
    secretKeyRef:
      name: rabbitmq
      key: username
  password:
    secretKeyRef:
      name: rabbitmq
      key: password
  exchangeConfig:
    type: "fanout"
    durable: true
    name: "my"
  queueConfig:
    name: "my"
    durable: true
    autoDelete: false
  channelConfig:
    parallelism: 1 
  predeclared: false
  sink:
    ref:
      apiVersion: v1
      kind: Service
      name: service-that-returns-error
  # Retry "forever" => 30 days" 
  retry: 2592000
  backoffPolicy: "linear"
  backoffDelay: "1s"
  
```
 **Expected behavior**
Message is retried

 **To Reproduce**
In sink section configure to service that returns error. Also configure retry policy (see example above)

 **Knative release version**
1.5.0

 **Additional context**
Also rabbitmq hangs up and source is not restarted, which results in broken rabbitmq eventing, so no messages can be delivered through

**Comment by gabo1208** [comment]
Thank you very much for reporting this issue @gunpuz !!

You are correct on the observation I'm finishing a patch to fix this (also there is a mayor source update announced for the source in the next release so really awesome that you report this :D!)!!

We were not using the correct `RetryConfig` in the `SendWithRetries` function in the `Adapter` so whenever a retry was been attempted the backoff function was null.

I'll let you know when the patch is ready 🚀 !

**Comment by gunpuz** [comment]
Thanks for the fast fixes @gabo1208. 👍  Any estimates (when/ in which version) this could be released?

**Comment by gabo1208** [comment]
No problem! And yes each fix gets cherry picked on Tuesdays so tomorrow or next Tuesday it will be available on release v1.5 :)
